java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/persistence/DynamoDbMetastoreImpl.java

```diff
@@ -42,21 +42,31 @@
 
   private final DynamoDB client;
   private final String tableName;
-  private final String keySuffix;
+  private final String preferredRegion;
+  private final Boolean hasKeySuffix;
   // Table instance can be cached since thread-safe and no state other than description, which we don't use
   private final Table table;
 
-  public static Builder newBuilder() {
-    return new Builder();
+  public static Builder newBuilder(final String region) {
+    return new Builder(region);
   }
 
   DynamoDbMetastoreImpl(final Builder builder) {
     this.client = new DynamoDB(builder.client);
     this.tableName = builder.tableName;
-    this.keySuffix = builder.keySuffix;
+    this.preferredRegion = builder.preferredRegion;
+    this.hasKeySuffix = builder.hasKeySuffix;
     this.table = client.getTable(tableName);
   }
 
+  private String getHashKey(final String key) {
```

Comment:
Probably add some comments regarding explanation of this private method. Does not seem very obvious.